### DISCUSSION TITLE
Canonical Ranking of Atoms

### DISCUSSION THREAD

**Discussion by pradipchm** [rdkit/rdkit#7062] [Q&A]
Hi,
I am using **rdmolfiles.CanonicalRankAtoms** for ranking the atoms in a molecule. I observed that two PDB structures of the same compound but different polymorphs collected from CCDC are giving different canonical rankings. Any suggestions on how to fix it or any other options to rank the atoms?

**Comment by bp-kelley** [comment]
If the atom or bond types are different, you, indeed, will might indeed get different canonical rankings.

If the PDBs are the same, can't you use residue names?

**Reply by pradipchm** [reply]
what do you mean by the PDBs are same?
I trying to match the atom sequence of two PDBs using atom ranking using this code https://gist.github.com/fabian-paul/abba9172d394dffb93624a710acbab16
Here are the two PDBs for the same compound but different polymorphs. 
1st one
HEADER    CSD ENTRY paba03
CRYST1   6.4341   4.6151  10.5313  90.00 100.73  90.00        Pn
SCALE1      0.155422  0.000000  0.029452       0.000000
SCALE2      0.000000  0.216680  0.000000       0.000000
SCALE3      0.000000  0.000000  0.096645       0.000000
HETATM    1  O2  UNK     1       4.357   3.198   4.682  1.00  3.53           O  
HETATM    2  O1  UNK     1       2.614   3.669   3.384  1.00  3.03           O  
HETATM    3  H1  UNK     1       3.179   4.117   2.992  1.00  4.42           H  
HETATM    4  C7  UNK     1       3.164   3.051   4.353  1.00  2.33           C  
HETATM    5  N1  UNK     1      -0.115  -0.756   7.026  1.00  3.65           N  
HETATM    6  H1A UNK     1      -0.741  -1.104   6.486  1.00  4.58           H  
HETATM    7  H1B UNK     1       0.389  -1.406   7.362  1.00  4.58           H  
HETATM    8  C2  UNK     1       1.017   1.799   4.600  1.00  3.02           C  
HETATM    9  H2  UNK     1       0.686   2.257   3.862  1.00  3.63           H  
HETATM   10  C3  UNK     1       0.224   0.859   5.248  1.00  2.35           C  
HETATM   11  H3  UNK     1      -0.637   0.688   4.943  1.00  2.84           H  
HETATM   12  C4  UNK     1       0.719   0.175   6.352  1.00  2.14           C  
HETATM   13  C5  UNK     1       2.007   0.431   6.808  1.00  2.53           C  
HETATM   14  H5  UNK     1       2.338  -0.027   7.547  1.00  3.08           H  
HETATM   15  C6  UNK     1       2.800   1.371   6.161  1.00  2.85           C  
HETATM   16  H6  UNK     1       3.662   1.542   6.466  1.00  3.40           H  
HETATM   17  C1  UNK     1       2.305   2.055   5.057  1.00  2.57           C  
CONECT    1    4
CONECT    2    3    4
CONECT    3    2
CONECT    4    1    2   17
CONECT    5    6    7   12
CONECT    6    5
CONECT    7    5
CONECT    8    9   10   17
CONECT    9    8
CONECT   10    8   11   12
CONECT   11   10
CONECT   12    5   10   13
CONECT   13   12   14   15
CONECT   14   13
CONECT   15   13   16   17
CONECT   16   15
CONECT   17    4    8   15
MASTER        0    0    0    0    0    0    0    3   17    0   17    0
END

2nd one 
HEADER    CSD ENTRY I
CRYST1   6.2782   8.5831  12.3649  90.00 100.13  90.00     P21/n
SCALE1      0.159281  0.000000  0.028467       0.000000
SCALE2      0.000000  0.116508  0.000000       0.000000
SCALE3      0.000000  0.000000  0.082156       0.000000
HETATM    1  O1  UNK     1      -0.566   2.983   4.994  1.00  4.15           O  
HETATM    2  H1  UNK     1       0.225   3.568   4.588  1.00  4.97           H  
HETATM    3  C1  UNK     1       0.027   2.611   6.127  1.00  3.04           C  
HETATM    4  O2  UNK     1       1.176   2.905   6.374  1.00  4.43           O  
HETATM    5  C2  UNK     1      -2.377   0.283   8.774  1.00  2.85           C  
HETATM    6  C3  UNK     1      -0.835   1.813   7.024  1.00  2.79           C  
HETATM    7  N1  UNK     1      -3.178  -0.433   9.685  1.00  3.51           N  
HETATM    8  H1A UNK     1      -2.643  -0.864  10.240  1.00  4.18           H  
HETATM    9  H1B UNK     1      -3.805  -1.025   9.273  1.00  4.18           H  
HETATM   10  C4  UNK     1      -2.079   1.325   6.625  1.00  3.10           C  
HETATM   11  H4  UNK     1      -2.338   1.548   5.594  1.00  3.71           H  
HETATM   12  C5  UNK     1      -1.154   0.798   9.189  1.00  3.39           C  
HETATM   13  H5  UNK     1      -0.833   0.636  10.248  1.00  4.11           H  
HETATM   14  C6  UNK     1      -2.840   0.557   7.489  1.00  3.20           C  
HETATM   15  H6  UNK     1      -3.736   0.085   7.133  1.00  3.87           H  
HETATM   16  C7  UNK     1      -0.393   1.548   8.321  1.00  3.31           C  
HETATM   17  H7  UNK     1       0.579   1.911   8.540  1.00  3.95           H  
CONECT    1    2    3
CONECT    2    1
CONECT    3    1    4    6
CONECT    4    3
CONECT    5    7   12   14
CONECT    6    3   10   16
CONECT    7    5    8    9
CONECT    8    7
CONECT    9    7
CONECT   10    6   11   14
CONECT   11   10
CONECT   12    5   13   16
CONECT   13   12
CONECT   14    5   10   15
CONECT   15   14
CONECT   16    6   12   17
CONECT   17   16
MASTER        0    0    0    0    0    0    0    3   17    0   17    0
END



 it does not give the same ranking. The ranking among Cs and Hs are different. Surprisingly it gives a similar ranking for a few other structures.

**Reply by bp-kelley** [reply]
If the proteins have different chemical graphs, the canonicalizations will be different.

Traditionally in pub files, monomer atoms are identified by their labels:

HETATM 16 C7

Here C7 is the atom label.  If these aren't retained between polymorphs, you'll have try other techniques to map the atoms together.

**Reply by pradipchm** [reply]
Do you have any suggestion for other techniques?
